### PR TITLE
chore(examples): update devcontainer-{docker,kubernetes} to use computed env

### PR DESCRIPTION
**Pull request by johnstcn**
Relates to https://github.com/coder/coder/issues/13278

### CODE REVIEW

File: examples/templates/devcontainer-kubernetes/main.tf [hunk 1]
```diff
@@ -260,49 +266,19 @@ resource "kubernetes_deployment" "main" {
...
-          env {
-            name  = "ENVBUILDER_DOCKER_CONFIG_BASE64"
-            value = try(data.kubernetes_secret.cache_repo_dockerconfig_secret[0].data[".dockerconfigjson"], "")
+
+          # Set the environment using cached_image.cached.0.env if the cache repo is enabled.
+          # Otherwise, use the local.envbuilder_env.
+          # You could alternatively write the environment variables to a ConfigMap or Secret
+          # and use that as `env_from`.
+          dynamic "env" {
+            for_each = nonsensitive(var.cache_repo == "" ? local.envbuilder_env : envbuilder_cached_image.cached.0.env_map)
```
Comment (johnstcn) [inline examples/templates/devcontainer-kubernetes/main.tf:275]: review: apparently this is required to be able to `for_each` over a `map` :thinking:

Comment (matifali) [reply examples/templates/devcontainer-kubernetes/main.tf:275]: Yes. I previously used it here. 
https://github.com/coder/modules/blob/4c45d69994937700f3033f120a489e1c17c1afd6/hcp-vault-secrets/main.tf#L69

Comment (mafredri) [reply examples/templates/devcontainer-kubernetes/main.tf:275]: This makes sense since `resource "envbuilder_cached_image" "cached"` has count set to 1.

**Review by mafredri — APPROVED** [review]
Nice cleanup!

**Pull request merged**